extension/builtin/default_registry.go

```diff
@@ -54,27 +53,45 @@ func (r *defaultRegistry) GetMapStorage() (storage.MapStorage, error) {
 	return mysql.NewMapStorage(r.db), nil
 }
 
-func (r *defaultRegistry) GetSigner(treeID int64) (*crypto.Signer, error) {
-	return r.signer, nil
+func (r *defaultRegistry) GetKeyProvider() (keys.Provider, error) {
+	return keyProvider{}, nil
 }
 
 // NewExtensionRegistry returns an extension.Registry implementation backed by a given
-// MySQL database and signer.
-func NewExtensionRegistry(db *sql.DB, signer *crypto.Signer) (extension.Registry, error) {
-	return &defaultRegistry{db: db, signer: signer}, nil
-
+// MySQL database. It supports loading private keys from PEM files.
+func NewExtensionRegistry(db *sql.DB) (extension.Registry, error) {
+	return &defaultRegistry{db: db}, nil
 }
 
 // NewDefaultExtensionRegistry returns the default extension.Registry implementation, which is
 // backed by a MySQL database and configured via flags.
+// It supports loading private keys from PEM files.
 func NewDefaultExtensionRegistry() (extension.Registry, error) {
 	db, err := mysql.OpenDB(*MySQLURIFlag)
 	if err != nil {
 		return nil, err
 	}
-	key, err := keys.NewFromPrivatePEMFile(*privateKeyFile, *privateKeyPassword)
-	if err != nil {
-		return nil, err
+	return NewExtensionRegistry(db)
+}
+
+// keyProvider implements keys.Provider.
+type keyProvider struct{}
```

Comment:
Consider moving this to a new file in the same folder for better readability. A kind of one class per file rule of thumb. 

Consider renaming to `signerFactory`